### PR TITLE
wait 5 seconds for the plugins to be fully loaded

### PR DESCRIPTION
**Pull request by MoonshineSG**
Wait 5 seconds for all the plugins to start (especially those handling alternative UI) before reloading the page.

### CODE REVIEW

File: src/octoprint/static/intermediary.html [hunk 1]
```diff
@@ -172,13 +176,15 @@
...
                         message.innerText = "OctoPrint server online, reloading page...";
-
-                        var reloadUrl = baseUrl + currentQuery + currentFragment;
-                        if (reloadUrl == window.location.href) {
-                            window.location.reload(false);
-                        } else {
-                            window.location.href = reloadUrl;
-                        }
+             
+                        sleep(5000).then(() => {
```
Comment (foosel) [inline src/octoprint/static/intermediary.html]: No ES6 syntax for now please :) Might introduce browser compatibility issues

**Comment by MoonshineSG** [comment]
I'm not sure why the build failed...

**Comment by foosel** [comment]
Not your fault, I broke something in the tests myself but haven't yet
gotten around to fixing it due to hunting down some other issue for the
past days.

**Comment by foosel** [comment]
Could you describe the problem you encountered a bit more for which this is the solution? Plugins should usually not need to "start up", the only thing that might not have happened yet is that the preemptive caching not having taken place.

UiPlugins in general shouldn't need time to startup though, either they can deliver a page or they can't, and introducing a general delay of 5s (or more, your current implementation sets a timeout of 10000ms instead of 5000) into the reload to mask a different issue (which I suspect is in the way here) is something I'd like to avoid :)

**Comment by MoonshineSG** [comment]
Yes, it's the preemptive caching. The plugin itself does not have any startup time needed. If the request comes before that (analysed from the log files) OctoPrint serves the default page without checking if there's a UIPlugin that matches the conditions for a different index page. 

I increased to 10sec (too much if you ask me) because I had few instances where 5sec still was not enough... 
I understand your reluctance to add 10sec to the startup time. Please let me know if there's an alternative. 

As you suggested, I am trying to change the passing of APIKEY from GET to POST. I wonder how would that work with the "intermediary.html" ?
For now I am having problems, with that as the root does not accept a POST method. But this is a different problem.

**Comment by foosel** [comment]
>  If the request comes before that (analysed from the log files) OctoPrint serves the default page without checking if there's a UIPlugin that matches the conditions for a different index page.

Definitely a bug then that should rather be addressed than adding arbitrary delays. I actually looked into that stuff about a week ago since I spotted a similar behaviour and thought I'd had solved that. Note that the preemptive cache is completely server side though, so it will still cache the wrong things potentially, even if the intermediary.html does the redirect at a later point in time. So I'm not entirely sure what you are observing there specifically, it would be great if you could give some steps to reproduce reliably.

> As you suggested, I am trying to change the passing of APIKEY from GET to POST

Sorry, you misunderstood me there. I meant for the apikey to not be supplied in the query but in a custom header, [as described here](http://docs.octoprint.org/en/master/api/general.html#authorization). I'm also still unsure why you need it at all for querying the index page (you do not need an apikey to access `/`, or it would be impossible to ever access that page, the api key is only needed for api access, not for accessing HTML pages), but I'll try to get back to you on that on hangouts asap.

**Comment by MoonshineSG** [comment]
Ok... I updated to version `dev1223_g7908f52` (i was on `dev1209_g9b6b4a9`) and not it seams that the problem is gone (without the delay from this pr). Seams like you crushed that bug.  Good news!

**Comment by foosel** [comment]
So that change isn't necessary for you anymore?

**Pull request closed without merging**